ref/general/zero-migration-architecture.adoc

```diff
@@ -0,0 +1,92 @@
+
+Open Liberty provides an absolute separation between product files and user files.
+Product files do not need to be modified by users.
+By default, user files such as server configuration and application files are located under a separate directory from the product files.
+If you want to override the default location, for example to put them on a separate file system from the product files you can set the `WLP_USER_DIR` environment variable.
+For more information, see link:/docs/ref/config/serverConfiguration.html[Server configuration].
+
+== Pluggable features
+
+Open Liberty features are pluggable.
+Your existing features are supported in new product versions.
+You can add, remove, and update features as needed, with minimal impact on your application.
+Open Liberty uses pluggable features to support multiple versions of an API.
+Changes in API behavior are enabled in new feature versions so that you can choose the appropriate feature version for your application.
+Open Liberty continues to support these versioned features across runtime updates.
+If you continue to use the same feature version, you never need to migrate your application.
+
+For example, Open Liberty supports both the servlet 3.0 and 3.1 specifications.
+The servlet 3.1 specification includes clarifications to previous servlet versions that can result in changes in application behavior.
+Open Liberty restricts any such behavior changes to the `servlet-3.1` feature and retains the existing behavior in the `servlet-3.0` feature.
+If your application is configured for the servlet 3.0 specification, you can continue to use the servlet-3.0 feature across runtime updates, regardless of how many other servlet specification levels are supported.
+You don't need to migrate your application unless you decide to configure the `servlet-3.1` feature instead.
+For more information, see link:/docs/ref/feature/#featureOverview.html[Server features].
+
+=== Considerations for using new features
+
+A new version of a feature that you are already using might impact your existing applications.
+For example, if you currently use the `servlet-3.0` feature and you want to use the `servlet-3.1` feature, your existing servlet applications might need to be changed to work correctly with servlet 3.1.
+Modify your application to work with the new feature version or keep your applications in a server that is configured with the original feature version and create a server configuration with the new version for new applications.
+
+Some features interact closely with other features and are sensitive to the feature versions when they are both configured into the same server.
+Some features require specific versions of prerequisite software, most commonly the Java SDK.
+Java EE 7 features, for example, require Java version 7 as a minimum.
+Therefore, adding a Java EE 7 feature to your server configuration might require you to use Java SDK 7 or later.
+
+== Exceptions to zero migration
```

Comment:
Zero migration was hyphenated at the beginning of the topic. Check this throughout for consistency.